packages/typespec-mcp/src/typekit/index.ts

```diff
@@ -63,7 +64,9 @@ defineKit<TypekitExtension>({
     tools: {
       list(server?: McpServer) {
         const toolState = this.program.stateMap(stateKeys.tool);
-        const allToolOps = Array.from(toolState.keys()) as Operation[];
+        const allToolOps = (Array.from(toolState.keys()) as Operation[]).filter(
```

Comment:
updated in this pr with the approach we talked about in the the chat https://github.com/microsoft/typespec-mcp/pull/13